webpack.config.js

```diff
@@ -0,0 +1,33 @@
+const path = require("path");
+
+module.exports = {
+    mode: "none",
+    entry: ["@babel/polyfill", "./lib/linter.js"],
+    output: {
+        filename: "eslint.js",
+        library: "eslint",
+        libraryTarget: "umd",
+        globalObject: "this"
+    },
+    module: {
+        rules: [
+            {
+                test: path.resolve("./lib/linter.js"),
+                loader: "string-replace-loader",
+                options: {
+                    search: "require(parserName)",
+                    replace: "(parserName === \"espree\" ? require(\"espree\") : require(parserName))"
```

Comment:
Could we make this replacement in the `lib/linter.js` source code itself rather than doing string replacement at build time? It seems like that would make the build process simpler (for us and also for anyone else bundling ESLint).

Alternatively, we could run `this.defineParser("espree", require("espree"))` in the `Linter` constructor.

Reply:
That's good! And we can remove "string-replace-loader" then.

Reply:
this could cause a little perf regression, if a user is using a different parser (eg. babel-eslint). I slightly prefer not to do this(if we can).

Reply:
Do you prefer the way which "string-replace-loader" does but implemented into `Linter`?

Reply:
maybe we can figure out another way to address @not-an-aardvark ‘s concern. I can try it later.

Reply:
https://github.com/eslint/eslint/blob/95aa3fdb392d265e6c3d813d54076458e88e7ad8/lib/linter.js#L448-L450
can we put something like
```js
if (parserName === DEFAULT_PARSER_NAME) {
  parser = parserMap.get("espree") || require("espree");
}
```
to lazyload the default parser? what do you think? @g-plane  @not-an-aardvark

Reply:
I have no objection.

Reply:
I don't think the perf regression is significant because it would only happen once globally. If we also want to avoid the time spent looking at the `require` cache, we could do something like this:

```js
// at the top of the file:
const espree = require("espree");

// in the Linter constructor:
this.defineParser("espree", espree);
```

In https://github.com/eslint/eslint/pull/11388, the dynamic loading is being removed from `Linter` anyway, so it uses this solution to always preload `espree`.